zebra-state/src/service/read/difficulty.rs

```diff
@@ -308,16 +319,28 @@ fn adjust_difficulty_and_time_for_testnet(
         .expect("a valid block time minus a small constant is in-range");
 
     if result.cur_time <= local_std_difficulty_limit {
-        // Standard difficulty: the max time needs to exclude min difficulty blocks
-        result.max_time = std_difficulty_max_time;
+        // Standard difficulty: the cur and max time need to exclude min difficulty blocks
+
+        // The maximum time can only be decreased, and only as far as min_time.
+        // The old minimum is still required by other consensus rules.
+        result.max_time = min(std_difficulty_max_time, result.max_time);
+        result.max_time = max(result.min_time, result.max_time);
+
+        // The current time only needs to be decreased if the max_time decreased past it.
+        // Decreasing the current time can't change the difficulty.
+        result.cur_time = result.cur_time.clamp(result.min_time, result.max_time);
```

Comment:
These fixes only matter if the last block's time is particularly large or small. (Which can happen, because the other consensus rules use the median time of recent blocks, not the tip block time.)